core/src/main/java/org/apache/accumulo/core/data/AbstractId.java

```diff
@@ -32,7 +32,7 @@ public abstract class AbstractId<T extends AbstractId<T>> implements Comparable<
   private final String canonical;
 
   protected AbstractId(final String canonical) {
-    this.canonical = Objects.requireNonNull(canonical, "canonical cannot be null");
+    this.canonical = Objects.requireNonNull(canonical, "canonical cannot be null").intern();
```

Comment:
Would not want to add `intern()` call here as it would impact all id types, like the Fate and Compaction id that are very high cardinality.